packages/components/src/Drawer/Drawer.component.js

```diff
@@ -21,14 +23,15 @@ class DrawerAnimation extends React.Component {
 	}
 
 	render() {
-		const { children, ...rest } = this.props;
+		const { children, withTransition, ...rest } = this.props;
+		const transitionDuration = withTransition ? DEFAULT_TRANSITION_DURATION : 0;
 		return (
 			<CSSTransition
 				{...rest}
 				onTransitionComplete={this.handleTransitionComplete}
 				defaultStyle={{ transform: 'translateX(100%)' }}
-				enterStyle={{ transform: transit('translateX(0%)', 350, 'ease-in-out') }}
-				leaveStyle={{ transform: transit('translateX(100%)', 350, 'ease-in-out') }}
+				enterStyle={{ transform: transit('translateX(0%)', transitionDuration, 'ease-in-out') }}
```

Comment:
What's about `enterStyle={withTransition && { transform: transit('translateX(0%)', DEFAULT_TRANSITION_DURATION, 'ease-in-out') }}`

Reply:
Isn't it better not to return `<CSSTransition>` if `withTransition` is false ?

Reply:
sadly not really, many forms with autorfocus rely on the onTransitionComplete hook, so some visual glitch don't happen.